### DISCUSSION TITLE
v3.1.0 Improved Triangle pattern detection and classification.

### DISCUSSION THREAD

**Discussion by BennyThadikaran** [BennyThadikaran/stock-pattern#39] [Announcements]
In previous versions, most Triangle patterns were being classified as Symmetrical with very few Ascending or Descending triangles.

This update fixes the issue so that Triangles are now properly classified. It also fixes a logical error in filtering to ensure Triangles have the correct shape and form.

**Breaking Change**: I fixed a typo. `Symetric` is changed to `Symmetric`. Because of this change, JSON files output from previous versions will throw an error on this version. Delete them and rescan.

**Comment by Prady04** [comment]
Guess this broke the windows fix again?

**Reply by Prady04** [reply]
![image](https://github.com/BennyThadikaran/stock-pattern/assets/71193926/69b501a1-df1e-46b6-ba13-6fc64fe17c92)

**Reply by BennyThadikaran** [reply]
Its neither a windows problem nor anything to do with this update. Its similar to the previous issue you raised. I thought it was fixed for both Double top and bottom. Anyways i know how to replicate this error. So will test it on all pattterns and hopefully fix it once and for all.

**Reply by BennyThadikaran** [reply]
Its fixed now in v3.1.1. Before scanning any of the patterns, i checked for duplicate indexes and removed them at the outset. This resolves the issue for all patterns. Thanks for being so forthcoming with issues. Really helps improve the code. :smiley:

**Comment by Prady04** [comment]
thank you for fixing it so fast. it works perfectly now. 
feedback on the new changes - many symmetrical triangles are getting marked as ascending and descending triangles now. 
for eg: please maxhealth, triveni , kajariacer, avantifeed as on 31-MAY-2024 eod2data

**Reply by BennyThadikaran** [reply]
Yes i agree. Due to logical errors prior to this update, i couldnt get a good fix on the correct slope parameters for the Triangle lines. I should be able to tighten the parameters. I will sit on it next Saturday.
